cli/src/cli_util.rs

```diff
@@ -2658,21 +2658,27 @@ fn resolve_default_command(
 
     if let Some(matches) = matches {
         if matches.subcommand_name().is_none() {
-            if config.get_string("ui.default-command").is_err() {
-                writeln!(
-                    ui.hint(),
-                    "Hint: Use `jj -h` for a list of available commands."
-                )?;
-                writeln!(
-                    ui.hint(),
-                    "Set the config `ui.default-command = \"log\"` to disable this message."
-                )?;
+            let mut default_command = match config.get("ui.default-command") {
+                Err(_) | Ok(CommandNameAndArgs::Structured { .. }) => {
+                    writeln!(
+                        ui.hint(),
+                        "Hint: Use `jj -h` for a list of available commands."
+                    )?;
+                    writeln!(
+                        ui.hint(),
+                        "Set the config `ui.default-command = \"log\"` to disable this message."
+                    )?;
+
+                    vec!["log".to_string()]
+                }
+                Ok(CommandNameAndArgs::String(name)) => vec![name],
+                Ok(CommandNameAndArgs::Vec(v)) => v.into(),
```

Comment:
To support space-separated string (like `"log -l1"`) for consistency, we'll have to use `.split_name_and_args()`. Perhaps, we can extract another split function that returns a slice instead of tuple.

If we don't need it, maybe it's easier to try both `.get_string()` and `.get::<Vec<String>>()`.

Reply:
Do you mean that we should have a method with this signature?

```rust
fn new_method(&self) -> Cow<[String]>;
```

Instead of the current:

```rust
pub fn split_name_and_args(&self) -> (Cow<str>, Cow<[String]>);
```

Reply:
> fn new_method(&self) -> Cow<[String]>;

Yes. Adding such method will help.

Another option is to try `.get_string()` and `.get::<Vec<String>>()` in order instead of using `CommandNameAndArgs` type. Since we can't support the `Structured` variant, it might be simpler.